Bonsai.Harp.CF/SyringePumpCommand.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Bonsai.Expressions;
+using System;
+using System.ComponentModel;
+using System.Linq.Expressions;
+
+namespace Bonsai.Harp.CF
+{
+    public enum SyringePumpCommandType : byte
+    {
+        EnableMotorDriver,
+        DisableMotorDriver,
+
+        StartProtocol,
+        StopProtocol,
+
+        SetDigitalOutputs,
+        ClearDigitalOutputs,
+
+        MotorMicrostep,
+
+        ProtocolNumberOfSteps,
+        ProtocolStepsPeriod,
+        ProtocolFlowRate,
+        ProtocolVolume,
+        ProtocolType,
+
+        CalibrationValue1,
+        CalibrationValue2
+    }
+
+    public enum MotorMicrostep : byte
+    {
+        Full = (0 << 0),
+        Half = (1 << 0),
+        Quarter = (2 << 0),
+        Eighth = (3 << 0),
+        Sixteenth = (4 << 0)
+    }
+
+    [TypeDescriptionProvider(typeof(DeviceTypeDescriptionProvider<SyringePumpCommand>))]
+    [Description("Creates standard command messages available to the Syringe Pump device")]
+    public class SyringePumpCommand : SelectBuilder, INamedElement
+    {
+        [RefreshProperties(RefreshProperties.All)]
+        [Description("Specifies which command to send to the Syringe Pump device.")]
+        public SyringePumpCommandType Type { get; set; } = SyringePumpCommandType.StartProtocol;
+
+        public MotorMicrostep Mask { get; set; } = MotorMicrostep.Full;
```

Comment:
Indeed, this approach with the `Mask` operator has been so far prevalent in several of the Harp devices. In reality, this is symptomatic of a deeper problem with the approach we have taken so far in designing the Bonsai interface for these nodes.

Basically the deeper discussion is whether we should have only one `Event` and `Command` node per device, or whether each device is really a namespace, with different nodes for each operation (or group of operations). There are pros and cons to each, but in general I've been leaning towards a more granular approach for recent devices.

Basically what is happening with the case of the `MotorMicrostep` command is that it really has an extra parameter. If operations each have their own node, then their parameter arguments can be exposed with more readable domain-specific names, and also they will have a description in the toolbox, etc.

This was not done initially to prevent the multiplication of nodes in the toolbox. However, since then two improvements have been discussed to prevent confusion. First, the nodes in the toolbox are searchable by namespace, so in this case if there was a `SyringePump` namespace, one could find all of its operations simply by typing `SyringePump` in the toolbox.

Second, the granularity of operations can be controlled. Basic operations such as `Start`/`Stop` and `Enable`/`Disable` could be grouped into one node for operation control. Other operations that require input might raise new questions about how to deal with inputs vs configuration values.

However, this deeper discussion is outside of the scope of this review, and for now this style is consistent with the rest of the package.

Reply:
Thanks for addressing some of the questions that I had regarding this part.